shared/AppInsightsCommon/Tests/Unit/src/AppInsightsCommon.tests.ts

```diff
@@ -293,5 +293,85 @@ export class ApplicationInsightsTests extends AITestClass {
+                Assert.equal(invalidUrl, result, "Invalid URLs should be returned unchanged");
+            }
+        });
+        this.testCase({
+            name: 'DataSanitizerTests: dataSanitizerUrl still enforces maximum length after redaction',
+            test: () => {
+                // Setup
+                const loggerStub = this.sandbox.stub(this.logger, "throwInternal");
+                const MAX_URL_LENGTH = DataSanitizerValues.MAX_URL_LENGTH;
+
+                // Create a very long URL with sensitive information
+                const longBaseUrl = "https://username:password@example.com/";
+                const longPathPart = strRepeat("a", MAX_URL_LENGTH);
+                const longUrl = longBaseUrl + longPathPart;
+
+                // Act
+                const result = dataSanitizeUrl(this.logger, longUrl);
+
+                // Assert
+                Assert.equal(MAX_URL_LENGTH, result.length, "URL should be truncated to maximum length");
+                Assert.equal(true, result.indexOf("REDACTED") > -1, "Redaction should happen before truncation");
+                Assert.ok(loggerStub.calledOnce, "Logger should be called once for oversized URL");
+
+                loggerStub.restore();
+            }
+        });
+        this.testCase({
+            name: 'DataSanitizerTests: dataSanitizerUrl handles null and undefined inputs',
+            test: () => {
+                // Act & Assert
+                const nullResult = dataSanitizeUrl(this.logger, null);
+                Assert.equal(null, nullResult, "Null input should return null");
+
+                const undefinedResult = dataSanitizeUrl(this.logger, undefined);
+                Assert.equal(undefined, undefinedResult, "Undefined input should return undefined");
+            }
+        });
+        this.testCase({
+            name: 'DataSanitizerTests: dataSanitizerUrl preserves URLs with no sensitive information',
+            test: () => {
+                // URL with no sensitive information
+                const safeUrl = "https://example.com/api?param1=value1&param2=value2";
+
+                // Act & Assert
+                const result = dataSanitizeUrl(this.logger, safeUrl);
+                Assert.equal(safeUrl, result, "URL with no sensitive info should remain unchanged");
+            }
+        });
+        this.testCase({
+            name: 'DataSanitizerTests: dataSanitizerUrl properly redacts sensitive query parameters',
+            test: () => {
+                // URLs with sensitive query parameters
+                const urlWithSensitiveParams = "https://example.com/api?Signature=secret&normal=value";
+                const expectedRedactedUrl = "https://example.com/api?Signature=REDACTED&normal=value";
+
+                // Act & Assert
+                const result = dataSanitizeUrl(this.logger, urlWithSensitiveParams);
+                Assert.equal(expectedRedactedUrl, result);
+            }
+        });
```

Comment:
Should we add unit tests for edge-cases such as empty string and non-string input?